### PR TITLE
New package: device-mapper

### PR DESCRIPTION
**Pull request by DarylGrunau**
This PR adds support to build/install device-mapper, a component of the linux kernel that supports logical volume management.

### CODE REVIEW

**Comment by hartzell** [comment]
I think that it's worth calling out somewhere (in the package doc string or ???) that this installs the *user space* components of the device mapper (dmsetup and libdevmapper), not the kernel componentry.

File: var/spack/repos/builtin/packages/device-mapper/package.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class DeviceMapper(AutotoolsPackage):
+    """The device-mapper is a component of the linux kernel (since version
+    2.6) that supports logical volume management.  This package provides
+    the userspace configuration tools (dmsetup), library (libdevmapper) and
+    header files."""
+
+    homepage = "https://www.sourceware.org/dm"
+    url      = "https://sourceware.org/pub/lvm2/releases/LVM2.2.03.05.tgz"
+
+    version('2.2.03.05', sha256='ca52815c999b20c6d25e3192f142f081b93d01f07b9d787e99664b169dba2700')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/device-mapper/package.py]: I think the package name is LVM2 and the version is 2.03.05. I would change the version numbers and add a `url_for_version`:

```python
def url_for_version(self, version):
    url = "https://sourceware.org/pub/lvm2/releases/LVM2.{0}.tgz"
    return url.format(version)
```

File: var/spack/repos/builtin/packages/device-mapper/package.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    version('2.2.03.04', sha256='f151f36fc0039997d2d9369b607b9262568b1a268afe19fd1535807355402142')
+    version('2.2.03.03', sha256='cedefa63ec5ae1b62fedbfddfc30706c095be0fc7c6aaed6fd1c50bc8c840dde')
+    version('2.2.03.02', sha256='550ba750239fd75b7e52c9877565cabffef506bbf6d7f6f17b9700dee56c720f')
+    version('2.2.03.01', sha256='424e58b074195ec08e0315fa1aff2550590998c33aea5c43bdceb8c1d135530b')
+    version('2.2.03.00', sha256='405992bf76960e60c7219d84d5f1e22edc34422a1ea812e21b2ac3c813d0da4e')
+
+    depends_on('libaio')
+
+    def configure_args(self):
+        return ['--enable-pkgconfig']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/device-mapper/package.py]: Does this require pkgconfig as a build dependency?

File: var/spack/repos/builtin/packages/device-mapper/package.py (hunk 1 above)
Comment (DarylGrunau) [reply var/spack/repos/builtin/packages/device-mapper/package.py]: agreed ...

File: var/spack/repos/builtin/packages/device-mapper/package.py (hunk 2 above)
Comment (DarylGrunau) [reply var/spack/repos/builtin/packages/device-mapper/package.py]: Yes, see new pushed content ...

File: var/spack/repos/builtin/packages/lvm2/package.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    version('2.03.03', sha256='cedefa63ec5ae1b62fedbfddfc30706c095be0fc7c6aaed6fd1c50bc8c840dde')
+    version('2.03.02', sha256='550ba750239fd75b7e52c9877565cabffef506bbf6d7f6f17b9700dee56c720f')
+    version('2.03.01', sha256='424e58b074195ec08e0315fa1aff2550590998c33aea5c43bdceb8c1d135530b')
+    version('2.03.00', sha256='405992bf76960e60c7219d84d5f1e22edc34422a1ea812e21b2ac3c813d0da4e')
+
+    def url_for_version(self, version):
+        url = "https://sourceware.org/pub/lvm2/releases/LVM2.{0}.tgz"
+        return url.format(version)
+
+    provides('device-mapper')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/lvm2/package.py]: Provides is intended for virtual dependencies that can be provided by multiple packages. I don't think that's the case here.

File: var/spack/repos/builtin/packages/lvm2/package.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    variant('pkgconfig', default=True,
+            description='install pkgconfig support')
+
+    depends_on('libaio')
+    depends_on('pkgconfig', type='build', when='+pkgconfig')
+
+    def configure_args(self):
+        args = ['--with-confdir={0}'.
+                format(join_path(self.prefix, 'etc')),
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/lvm2/package.py]: You can do `self.prefix.etc`

File: var/spack/repos/builtin/packages/lvm2/package.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+                '--with-default-system-dir={0}'.
+                format(join_path(self.prefix, 'etc', 'lvm'))]
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/lvm2/package.py]: You can do `self.prefix.etc.lvm`

File: var/spack/repos/builtin/packages/lvm2/package.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        if self.spec.satisfies('+pkgconfig'):
+            args.append('--enable-pkgconfig')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/lvm2/package.py]: I would explicitly disable this in an `else:` clause

File: var/spack/repos/builtin/packages/lvm2/package.py (hunk 3 above)
Comment (DarylGrunau) [reply var/spack/repos/builtin/packages/lvm2/package.py]: Ok, removed

File: var/spack/repos/builtin/packages/lvm2/package.py (hunk 6 above)
Comment (DarylGrunau) [reply var/spack/repos/builtin/packages/lvm2/package.py]: See new pushed content ...

**Pull request merged**